dotCMS/src/integration-test/java/com/dotcms/vanityurl/business/VanityUrlAPITest.java

```diff
@@ -60,271 +61,197 @@ public static void prepare() throws Exception {
         /* Default user */
         user = APILocator.systemUser();
 
-        /* APIs initialization */
-        hostAPI = APILocator.getHostAPI();
         contentletAPI = APILocator.getContentletAPI();
         vanityUrlAPI = APILocator.getVanityUrlAPI();
         filtersUtil = FiltersUtil.getInstance();
 
-        /* Load Cache */
-        vanityUrlCache = CacheLocator.getVanityURLCache();
+        cache=CacheLocator.getVanityURLCache();
 
         /* Default variables */
-        defaultHost = hostAPI.findDefaultHost(user, false);
-        defaultLanguageId = APILocator.getLanguageAPI().getDefaultLanguage().getId();
+        defaultHost =  new SiteDataGen().nextPersisted();
+        defaultLanguage = APILocator.getLanguageAPI().getDefaultLanguage();
     }
 
+    
+    
     /**
      * Creates a vanity url but no publish it, gets it and checks there is not in cache.
      *
      * @see VanityUrlAPI
      */
     @Test
-    public void getWorkingVanityUrl() {
+    public void testing_the_from_contentlet_method() throws Exception {
 
-        Contentlet contentlet1 = null;
 
-        try {
-            long i = System.currentTimeMillis();
-            String title = "VanityURL" + i;
-            String site = defaultHost.getIdentifier();
-            String uri = "/test1_" + i;
-            String forwardTo = "/about-us/" + CMSFilter.CMS_INDEX_PAGE;
-            int action = 200;
-            int order = 1;
-
-            contentlet1 = filtersUtil.createVanityUrl(title, site, uri,
-                    forwardTo, action, order, defaultLanguageId);
-
-            VanityUrl vanity1 = vanityUrlAPI.getVanityUrlFromContentlet(contentletAPI.find(contentlet1.getInode(), user, false));
-
-            Assert.assertNotNull(vanity1);
-            Assert.assertNotEquals(VanityUrlAPI.CACHE_404_VANITY_URL,vanity1.getInode());
-            Assert.assertEquals(title, vanity1.getTitle());
-            Assert.assertEquals(site, vanity1.getSite());
-            Assert.assertEquals(uri, vanity1.getURI());
-            Assert.assertEquals(forwardTo, vanity1.getForwardTo());
-            Assert.assertEquals(action, vanity1.getAction());
-
-            CachedVanityUrl vanityCached = vanityUrlAPI.getLiveCachedVanityUrl(uri, defaultHost, defaultLanguageId, user);
-
-            Assert.assertEquals(VanityUrlAPI.CACHE_404_VANITY_URL,vanityCached.getVanityUrlId());
-
-        } catch (DotDataException | DotSecurityException e) {
-            e.printStackTrace();
-            fail();
-        } finally {
-            try {
-                if (contentlet1 != null) {
-                    contentletAPI.destroy(contentlet1, user, false );
-                }
-            } catch (Exception e) {
-                Logger.error(this.getClass(), "Error cleaning up Vanity Url Links");
-            }
-        }
-    }
+        long i = System.currentTimeMillis();
+        String title = "VanityURL" + i;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "VanityURL" appears 4 times in this file; the first occurrence is on line 89](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)